frontend/src/Onboarding/TransferCoursesScreen.tsx

```diff
@@ -0,0 +1,363 @@
+  font-size: 12px;
+  margin-left: 4px;
+  margin-top: 16px;
+  margin-bottom: 8px;
+  font-weight: 500;
+`;
+
+const CourseWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`;
+
+const CourseText = styled.p`
+  font-size: 12px;
+  margin: 1px;
+  font-weight: 400;
+`;
+
+const ScrollWrapper = styled.div`
+  &::-webkit-scrollbar-thumb:hover {
+    background-color: rgba(0, 0, 0, 0.5);
+  }
+  &::-webkit-scrollbar-thumb {
+    background-clip: padding-box;
+    background-color: #c1c1c1;
+    border-color: transparent;
+    border-radius: 9px 8px 8px 9px;
+    border-style: solid;
+    border-width: 3px 3px 3px 4px;
+    box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
+  }
+  &::-webkit-scrollbar {
+    -webkit-appearance: none;
+    width: 16px;
+  }
+  &::-webkit-scrollbar-track:vertical {
+    border-left: 1px solid #e7e7e7;
+    box-shadow: 1px 0 1px 0 #f6f6f6 inset, -1px 0 1px 0 #f6f6f6 inset;
+  }
+`;
+
+/**
+ * Flattens the Requirement[] into only a list of Requirements/Requirement sets
+ * This means that all inner lists will only contain one class or a list of the primary class and its labs/recitations
+ * @param reqs
+ */
+function flatten(reqs: Requirement[]): IRequiredCourse[][] {
+  return reqs.map(flattenOne).reduce((array, cur) => array.concat(cur), []);
+}
+
+function flattenOne(req: Requirement): IRequiredCourse[][] {
+  if (req.type === "COURSE") {
+    return [[req as IRequiredCourse]];
+  } else if (
+    req.type === "AND" &&
+    req.courses.filter(c => c.type === "COURSE").length
+  ) {
+    return [req.courses as IRequiredCourse[]];
+  } else if (req.type === "AND" || req.type === "OR") {
+    return flatten(req.courses);
+  } else {
+    return [];
+  }
+}
+
+interface TransferCoursesScreenProps {
+  major: Major;
+  completedRequirements: IRequiredCourse[];
+  setCompletedCourses: (completedCourses: ScheduleCourse[]) => void;
+  setTransferCourses: (transferCourses: ScheduleCourse[]) => void;
+}
+
+type Props = TransferCoursesScreenProps & RouteComponentProps;
+
+interface State {
+  selectedRequirements: IRequiredCourse[];
+  completedNonTransfer: IRequiredCourse[];
+  modalVisible: boolean;
+  otherCourses: IRequiredCourse[][];
+}
+
+class TransferCoursesComponent extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      modalVisible: false,
+      selectedRequirements: [],
+      otherCourses: [],
+      completedNonTransfer: props.completedRequirements,
+    };
+  }
+
+  hideModal() {
+    this.setState({ modalVisible: false });
+  }
+
+  showModal() {
+    this.setState({ modalVisible: true });
+  }
+
+  // Adds the "other courses" to the state in the form of a IRequiredCourse[][] so that they can be
+  // processed by renderAllCourses
+  addOtherCourses(courses: ScheduleCourse[]) {
+    let reqCourseMap = courses.map((course: ScheduleCourse) => [
+      {
+        type: "COURSE",
+        classId: +course.classId,
```

Comment:
Is this "+" doing anything here in particular?

Reply:
It converts the string into a number because in IRequiredCourse it's a number